R/sort_ard_hierarchical.R

```diff
@@ -71,6 +71,14 @@ sort_ard_hierarchical <- function(x, sort = c("descending", "alphanumeric")) {
 
   x_args <- attributes(x)$args
   by_cols <- if (length(x_args$by) > 0) paste0("group", seq_along(length(x_args$by)), c("", "_level")) else NULL
+
+  # for calculations by highest severity, innermost variable is extracted from by
+  if (length(x_args$by) > 1) {
+    x_args$variables <- c(x_args$variables, x_args$by[-1])
+    x_args$include <- c(x_args$include, x_args$by[-1])
+    x_args$by <- x_args$by[-1]
+  }
```

Comment:
Can you help me understand why this is needed? I thought we'd just add the sort function at the bottom and be done 😆

Reply:
I think they're just minor bugs that we didn't encounter previously - this is to account for ARDs by highest severity, which means the last hierarchy level is specified in `by` instead of `variables` which affects the sorting.

Reply:
gotcha! thanks!